yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -282,8 +282,16 @@ export class Sequencer {
     const allTxs = [...txs, ...times(emptyTxCount, () => emptyTx)];
     this.log(`Building block ${globalVariables.blockNumber}`);
 
-    const [block] = await this.blockBuilder.buildL2Block(globalVariables, allTxs, newL1ToL2Messages);
-    return block;
+    // If the block is invalid, drop the transactions
+    try {
+      const [block] = await this.blockBuilder.buildL2Block(globalVariables, allTxs, newL1ToL2Messages);
+      return block;
+    } catch (err) {
+      const hashes = txs.map(tx => tx.hash);
```

Comment:
this is dropping all transactions, even if only one of them is offending?

Reply:
Currently we were only building blocks of 1 tx for the e2e, so should not be an issue. But I'm looking at it now because It gave other issues for the race in p2p test. 
Figured that the emission of same nullifier was not caught before the block building by the sequencer so it was unclear exactly what was offending, but adding a check earlier on.